app.py

```diff
@@ -4,9 +4,18 @@
 app = Flask(__name__)
 app.config.from_object('flask_config.Config')
 
-@app.route('/')
+@app.route('/', methods = ['POST', 'GET', 'PUT', 'DELETE'])
 def index():
-    return 'Hello World!'
+    if request.method == 'POST':
+        NewItem = request.form["NewItem"]
+        session.add_item(NewItem)
+        return render_template('index.html', items = session.get_items())
```

Comment:
You should instead `return redirect('/')`

The main reasons being so that you don't duplicate code and so that if the user refreshes the webpage, it will not resubmit the POST request.